architecture/core.rst

```diff
@@ -533,6 +559,51 @@ interrupt occurs:
             }
     }
 
+**Using device tree**
+
+Device drivers can rely on Device Tree to get references to the
+interrupt they consume. Refer to the Device Tree documentation
+for how to properly define interrupt controller chip devices and reference
+interrupts in consumer device nodes.
+
+The example in the previous section implementing an interrupt consumer driver
+could be implemented as below, when the platform uses Device Tree means
+to define some platform resources.
+
+.. code-block:: c
+
+    static TEE_Result foo_probe(const void *fdt, int node, const void *compat)
```

Comment:
Wouldn't it be better with an example in the code where it's implemented?

Reply:
I think this snippet illustrate well hiw it's expected to be used.
That said, you're right,n a real example would be better.
I have stm32 candidate drivers but they have not yet moved to this API.
Would you prefer this doc change to wait those are ready?

Reply:
I guess it's OK for now, as long as it compiles if copied.